### ISSUE TITLE
Callback.AddDynamic

### ISSUE THREAD

**Issue by dobby5** [getnamo/socketio-client-ue4#143]
Hello,

how do the callbacks work?

I have tried this but without success.

Socket->OnConnectedCallback.AddDynamic(this, &GameInstance::OnConnectComplete);

Can someone please help me there.

**Comment by getnamo** [comment]
Is this on a USocketIOClientComponent?

**Comment by dobby5** [comment]
I used FSocketIONative. In addition, the client establishes connections again and again. Although it is already connected.

**Comment by getnamo** [comment]
Aha, while the USocketIOClientComponent uses the event delegate system, the slightly lower level FSocketIONative uses raw TFunctions for its callback methods (you can and I recommend using C++ lambdas to bind to them). The component actually uses the FSocketIONative under the hood and you can see e.g. https://github.com/getnamo/socketio-client-ue4/blob/master/Source/SocketIOClient/Private/SocketIOClientComponent.cpp#L128 for an example on how OnConnected is implemented (signatures are defined at: https://github.com/getnamo/socketio-client-ue4/blob/master/Source/SocketIOClient/Public/SocketIONative.h#L55). In your case you would want to do something like:

```c++
#include "LambdaRunnable.h"
//...

Socket->OnConnectedCallback = [this](const FString& InSessionId)
{
	//Native callbacks apart from OnEvent are not guaranteed to be on the game thread, we need this wrapper for safe handling
	FLambdaRunnable::RunShortLambdaOnGameThread([this, InSessionId]
	{
		//`this` context is captured so we can call our function directly as if in a member function
		OnConnectComplete();
	});
};
```

Regarding the re-connection issue, it seems to be https://github.com/getnamo/socketio-client-ue4/issues/126 which I'm still looking for time to solve. A workaround if you use only windows/linux platform is to rollback to https://github.com/getnamo/socketio-client-ue4/releases/tag/0.10.0 which doesn't have the issue (other platforms are only available with the v1.0 architecture and will need #126 fix)

**Comment by dobby5** [comment]
Thx you. You're doing a great job. Thanks for the plugin
